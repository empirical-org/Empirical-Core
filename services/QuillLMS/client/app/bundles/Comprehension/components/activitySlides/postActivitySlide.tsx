import * as React from "react";

export const PostActivitySlide = ({ responses, user }) => {

  function handleClick() {
    window.location.href = "/"
  }

  function getRevisionCount() {
    let count = 0;
    if(!responses) { return '0 revisions' }
    Object.keys(responses).map(key => {
      count += responses[key].length;
    });
    if(count <= 3) { return '0 revisions' }
    // we subtract 3 to account for a student having 3 optimal first attempts
    const calculatedCount = count - 3;
    if(calculatedCount === 1) {
      return '1 revision';
    }
    return `${calculatedCount} revisions`;
  }

  const completionText = user ? `${user}, you completed the activity!` : 'You completed the activity!'

  return(
    <div className="post-activity-slide-container">
      <section id="information-section">
        <p className="slide-sub-text" id="first-sub-text">{completionText}</p>
        <p id="revision-text">{`You made ${getRevisionCount()}!`}</p>
        <img alt="An illustration of a party popper" id="celebration-vector" src={`${process.env.CDN_URL}/images/comprehension/celebrating-activity-completion.svg`} />
        <section id="reminder-badge-section">
          <img alt="An illustration of an A+ that is crossed out" id="grade-badge" src={`${process.env.CDN_URL}/images/comprehension/no-grade-badge.svg`} />
          <section id="reminder-text-section">
            <p id="reminder-header">Reminder about grades</p>
            <p id="reminder-subtext">Your teacher will see all of your revisions, but this activity was for practice, so it isnâ€™t graded.</p>
          </section>
        </section>
        <p className="slide-sub-text" id="second-sub-text">Revising is a sign of being a thoughtful writer. Be proud of the work you did today and celebrate your success!</p>
      </section>
      <section id="button-container">
        <button className="quill-button large secondary outlined focus-on-dark" onClick={handleClick} type="submit">Save and exit</button>
      </section>
    </div>
  );
}

export default PostActivitySlide;
