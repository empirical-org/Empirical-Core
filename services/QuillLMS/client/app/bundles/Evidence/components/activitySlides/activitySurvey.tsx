import * as React from 'react';

import useFocus from '../../../Shared/hooks/useFocus';
import { TrackAnalyticsEvent } from "../../actions/analytics";
import { Events } from '../../modules/analytics';

const baseImgSrc = `${process.env.CDN_URL}/images/pages/evidence`
const checkMarkSrc = `${baseImgSrc}/components-selection-controls-dark-enabled-selected.svg`

const SURVEY_QUESTION = "What did you think of this activity?"

const emojis = [
  {
    alt: 'Angry face',
    src: `${baseImgSrc}/angry-face.svg`
  },
  {
    alt: 'Frowning face',
    src: `${baseImgSrc}/frowning-face.svg`
  },
  {
    alt: 'Neutral face',
    src: `${baseImgSrc}/neutral-face.svg`
  },
  {
    alt: 'Smiling face',
    src: `${baseImgSrc}/smiling-face.svg`
  },
  {
    alt: 'Excited face',
    src: `${baseImgSrc}/excited-face.svg`
  }
]

const positiveMultipleChoiceOptions = [
  'I thought the feedback was helpful',
  'I thought completing the sentences was a good challenge',
  'I thought completing the sentences was easy',
  'I had no trouble reading the text',
  'I thought this topic was interesting'
]

const negativeMultipleChoiceOptions = [
  'I got stuck on the feedback',
  'I thought completing the sentences was too difficult',
  'I thought completing the sentences was too easy',
  'I thought the text was too difficult',
  'I thought this topic was boring'
]

const EmojiButton = ({ src, alt, emojiNumber, setSelectedEmoji, selectedEmoji, }) => {
  function handleClick() { setSelectedEmoji(emojiNumber) }

  return (
    <button className={`emoji-button ${selectedEmoji === emojiNumber ? 'selected' : ''}`} onClick={handleClick} type="button">
      <img alt={alt} src={src} />
    </button>
  )
}

const MultipleChoiceOption = ({ text, selectedMultipleChoiceOptions, setSelectedMultipleChoiceOptions, }) => {
  const optionIsSelected = selectedMultipleChoiceOptions.find(opt => opt === text)

  function handleCheckboxClick() {
    const newSelectedMultipleChoiceOptions = optionIsSelected ? selectedMultipleChoiceOptions.filter(opt => opt !== text) : selectedMultipleChoiceOptions.concat(text)
    setSelectedMultipleChoiceOptions(newSelectedMultipleChoiceOptions)
  }

  let checkbox = <span aria-label="Unchecked checkbox" className="quill-checkbox unselected" />
  if (optionIsSelected) {
    checkbox = (<span aria-label="Checked checkbox" className="quill-checkbox selected" >
      <img alt="Checked checkbox" src={checkMarkSrc} />
    </span>)
  }

  return (
    <button className={`multiple-choice-option ${optionIsSelected ? 'selected' : ''}`} onClick={handleCheckboxClick} type="button">
      {checkbox}
      <span className="text-container">{text}</span>
    </button>
  )
}

const ActivitySurvey = ({ activity, dispatch, sessionID, saveActivitySurveyResponse, setSubmittedActivitySurvey, }) => {
  const [selectedEmoji, setSelectedEmoji] = React.useState(null)
  const [selectedMultipleChoiceOptions, setSelectedMultipleChoiceOptions] = React.useState([])

  const [containerRef, setContainerFocus] = useFocus()

  React.useEffect(() => {
    setContainerFocus()
  }, [])

  React.useEffect(() => { setSelectedMultipleChoiceOptions([]) }, [selectedEmoji])

  function mapMultipleChoiceOptionsForEventParams() {
    const options = {}
    positiveMultipleChoiceOptions.forEach(option => {
      const key = option.split(' ').join('_')
      options[key] = false
    })
    negativeMultipleChoiceOptions.forEach(option => {
      const key = option.split(' ').join('_')
      options[key] = false
    })
    selectedMultipleChoiceOptions.forEach(option => {
      const key = option.split(' ').join('_')
      options[key] = true
    });
    return options
  }

  function handleSend() {
    if (!selectedMultipleChoiceOptions.length) { return }
    const activitySurveyResponse = {
      emoji_selection: selectedEmoji,
      multiple_choice_selections: selectedMultipleChoiceOptions,
      survey_question: SURVEY_QUESTION,
    }
    const trackingParams = {
      activity_name: activity?.title,
      tool_name: "Reading",
      rating: selectedEmoji,
      ...mapMultipleChoiceOptionsForEventParams()
    }
    const callback = () => setSubmittedActivitySurvey(true)
    dispatch(TrackAnalyticsEvent(Events.STUDENT_RATED_AN_ACTIVITY, trackingParams))
    saveActivitySurveyResponse({ sessionID, activitySurveyResponse, callback, })
  }

  function handleLinkClick() { window.location.href = process.env.DEFAULT_URL }

  let activitySurveyHeader
  let multipleChoiceSection

  if (selectedEmoji) {
    const multipleChoiceOptions = selectedEmoji > 3 ? positiveMultipleChoiceOptions : negativeMultipleChoiceOptions
    const multipleChoiceElements = multipleChoiceOptions.map(opt => <MultipleChoiceOption key={opt} selectedMultipleChoiceOptions={selectedMultipleChoiceOptions} setSelectedMultipleChoiceOptions={setSelectedMultipleChoiceOptions} text={opt} />)
    multipleChoiceSection = (<section>
      <h1>Select one or more options:</h1>
      {multipleChoiceElements}
    </section>)
  } else {
    activitySurveyHeader = (<section>
      <p>We need your help</p>
      <h1>{SURVEY_QUESTION}</h1>
    </section>)
  }

  const emojiButtons = emojis.map((emoji, i) => (
    <EmojiButton
      alt={emoji.alt}
      emojiNumber={i + 1}
      key={emoji.alt}
      selectedEmoji={selectedEmoji}
      setSelectedEmoji={setSelectedEmoji}
      src={emoji.src}
    />)
  )

  const sendButtonClassName = selectedMultipleChoiceOptions.length ? 'quill-button large secondary outlined focus-on-dark' : 'quill-button large disabled contained focus-on-dark'

  return (
    <div className="activity-follow-up-container activity-survey-container no-focus-outline" ref={containerRef} tabIndex={-1}>
      <div className="activity-survey-content">
        {activitySurveyHeader}
        <div className="emoji-buttons">{emojiButtons}</div>
        {multipleChoiceSection}
      </div>
      <div className="button-section">
        <a className="focus-on-dark" href={process.env.DEFAULT_URL} onClick={handleLinkClick}>Skip</a>
        <button className={sendButtonClassName} onClick={handleSend} type="button">Send</button>
      </div>
    </div>
  )
}

export default ActivitySurvey
