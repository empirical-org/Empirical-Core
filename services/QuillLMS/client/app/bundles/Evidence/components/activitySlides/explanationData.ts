export const explanationData = {
  1: {
    buttonText: "Next",
    header: "All writers revise",
    imageData: {
      imageAlt: "An illustration of a pencil circled with a revision",
      imageUrl: "images/evidence/revising-pencil.svg"
    },
    isBeta: false,
    step: 2,
    subtext: "Your goal is to practice and improve your writing skills.<br /><br />We will provide you with a lot of feedback on your writing. Youâ€™ll be able to revise each sentence up to five times so that you can become a stronger writer."
  },
  2: {
    buttonText: "Next",
    header: "We use a feedback bot",
    imageData: {
      imageAlt: "An illustration of a smiling robot head",
      imageUrl: "images/evidence/feedback-bot.svg"
    },
    isBeta: true,
    step: 3,
    subtext: "We use artificial intelligence (AI) to help us give you feedback on your writing. You should know that AI isn't always correct."
  },
  3: {
    buttonText: "Start",
    header: "Your teacher will see your score and writing",
    imageData: {
      imageAlt: "An illustration of a report page with a magnifying glass over it",
      imageUrl: "images/evidence/evidence-report.svg"
    },
    isBeta: false,
    step: 4,
    subtext: "Once the activity is complete, both you and your teacher will see a report with your score and your responses for each prompt."
  },
}
