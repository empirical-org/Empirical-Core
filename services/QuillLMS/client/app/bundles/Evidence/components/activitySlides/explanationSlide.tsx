import * as React from "react";

import Footer from './footer';

import useFocus from '../../../Shared/hooks/useFocus';

export const ExplanationSlide = ({ slideData, onHandleClick }) => {
  const { buttonText, header, imageData, isBeta, step, subtext } = slideData;
  const { imageAlt, imageUrl } = imageData;

  const [containerRef, setContainerFocus] = useFocus()

  React.useEffect(() => {
    setContainerFocus()
  }, [slideData])

  return(
    <div className="explanation-slide-container">
      <section className="no-focus-outline" id="information-section" ref={containerRef} tabIndex={-1}>
        <p aria-hidden={true} className="subtext">Good to know</p>
        <section id="header-container">
          <h1>{header}</h1>
          {isBeta && <div id="beta-tag">BETA</div>}
        </section>
        <img alt={imageAlt} id="image" src={`${import.meta.env.VITE_PROCESS_ENV_CDN_URL}/${imageUrl}`} />
        <p className="subtext">{subtext}</p>
      </section>
      <Footer buttonText={buttonText} onHandleClick={onHandleClick} step={step} />
    </div>
  );
}

export default ExplanationSlide;
