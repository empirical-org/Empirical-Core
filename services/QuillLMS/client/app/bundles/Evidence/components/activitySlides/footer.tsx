import * as React from "react";

export const Footer = ({ buttonText, onHandleClick, step }) => {

  function handleClick() {
    onHandleClick();
  }

  return(
    <section id="button-container">
      <span />
      <button className="quill-button large secondary outlined focus-on-dark" onClick={handleClick} type="submit">{buttonText}</button>
      <div id="step-indictator-container">
        <div className={`step-indicator ${step === 1 ? 'active' : ''}`} />
        <div className={`step-indicator ${step === 2 ? 'active' : ''}`} />
        <div className={`step-indicator ${step === 3 ? 'active' : ''}`} />
        <div className={`step-indicator ${step === 4 ? 'active' : ''}`} />
      </div>
    </section>
  );
}

export default Footer;
