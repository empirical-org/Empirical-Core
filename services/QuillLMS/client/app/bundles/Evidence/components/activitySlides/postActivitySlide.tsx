import * as React from "react";

import { BECAUSE, BUT, SO } from '../../../Shared/index'
import { PromptInterface } from "../../../Staff/interfaces/evidenceInterfaces";

const FIRST_STRONG_EXAMPLE = 'first_strong_example';
const SECOND_STRONG_EXAMPLE = 'second_strong_example';
const conjunctions = [BECAUSE, BUT, SO];

interface PostActivitySlideProps {
  handleClick: () => void,
  prompts: PromptInterface[],
  responses: any
}

export const PostActivitySlide = ({ handleClick, prompts, responses }: PostActivitySlideProps) => {

  function getStrongExemplar(conjunction: string, property: string) {
    const prompt = prompts.filter(prompt => prompt.conjunction === conjunction)[0];
    if (!prompt) { return }
    const stemWithoutConjunction = prompt.text.replace(conjunction, '');
    const exemplar = prompt[property];
    return <p>{stemWithoutConjunction}<u>{conjunction}</u>{exemplar}</p>;
  }

  function getResponsesForConjunction(conjunction: string) {
    const responsesKeyHash = {}
    const responsesKeys = Object.keys(responses);
    conjunctions.forEach((conjunctionString, i) => responsesKeyHash[conjunctionString] = responsesKeys[i]);
    return responses[responsesKeyHash[conjunction]];
  }

  function getResponse(conjunction: string) {
    const responses = getResponsesForConjunction(conjunction)
    const lastResponseText = responses[responses.length - 1].entry;
    const splitResponse = lastResponseText.split(conjunction);
    return <p>{splitResponse[0]}<u>{conjunction}</u>{splitResponse[1]}</p>
  }

  function getRevisionCopy() {
    let count = 0;
    const noRevisionsCopy = 'You completed the activity with 0 revisions!';
    if(!responses) { return noRevisionsCopy }
    Object.keys(responses).map(key => {
      count += responses[key].length;
    });
    if(count <= 3) { return noRevisionsCopy }
    // we subtract 3 to account for a student having 3 optimal first attempts
    const calculatedCount = count - 3;
    if(calculatedCount === 1) {
      return 'You completed the activity with 1 revision!';
    }
    return `You completed the activity with ${calculatedCount} revisions!`;
  }

  function renderResponseAndExamplarsSection(conjunction: string) {
    return(
      <section className="response-exemplars-section">
        <section className="response-section">
          <div className="response-exemplar-header">
            <p className="sub-header-text">You wrote</p>
          </div>
          <section className="response-box">{getResponse(conjunction)}</section>
        </section>
        <section className="exemplars-section">
          <div className="response-exemplar-header">
            <p className="sub-header-text">Example Responses</p>
          </div>
          <section className="exemplar-box">{getStrongExemplar(conjunction, FIRST_STRONG_EXAMPLE)}</section>
          <section className="exemplar-box">{getStrongExemplar(conjunction, SECOND_STRONG_EXAMPLE)}</section>
        </section>
      </section>
    );
  }

  return(
    <div className="post-activity-slide-container">
      <section id="information-section">
        <img alt="An illustration of a party popper" id="celebration-vector" src={`${process.env.CDN_URL}/images/evidence/party-celebration.svg`} />
        <p id="revision-text">{getRevisionCopy()}</p>
        <p className="slide-sub-text" id="second-sub-text">Revising is a sign of being a thoughtful writer. Be proud of the work you did today, and celebrate your success!</p>
        <section id="reminder-badge-section">
          <img alt="An illustration of an A+ that is crossed out" id="grade-badge" src={`${process.env.CDN_URL}/images/evidence/paper-check.svg`} />
          <section id="reminder-text-section">
            <p className="sub-header-text">Reminder about grades</p>
            <p className="sub-header-subtext">Your teacher will see all of your revisions, but this activity was for practice, so it isnâ€™t graded.</p>
          </section>
        </section>
      </section>
      <section className="responses-exemplars-container">
        <section className="review-response-header-section">
          <p className="sub-header-text">Review the other responses.</p>
          <p className="sub-header-subtext">How is your response similar or different from these responses?</p>
        </section>
        {renderResponseAndExamplarsSection(BECAUSE)}
        {renderResponseAndExamplarsSection(BUT)}
        {renderResponseAndExamplarsSection(SO)}
      </section>
      <section id="button-container">
        <button className="quill-button large secondary outlined focus-on-dark" onClick={handleClick} type="button">Next</button>
      </section>
    </div>
  );
}

export default PostActivitySlide;
