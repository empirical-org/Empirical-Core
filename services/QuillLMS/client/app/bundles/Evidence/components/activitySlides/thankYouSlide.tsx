import * as React from 'react'

const applaudingSrc = `${process.env.CDN_URL}/images/pages/evidence/applauding.svg`

const ThankYouSlide = () => {
  const backToDashboard = () => {
    window.location.href = process.env.DEFAULT_URL
  }

  return (<div className="activity-follow-up-container thank-you-slide-container">
    <div className="thank-you-content">
      <section>
        <p>Your feedback was sent</p>
        <h1>Thanks for sharing your feedback! </h1>
      </section>
      <img alt="Two hands clapping together" src={applaudingSrc} />
    </div>
    <div className="button-section">
      <a className='quill-button large secondary outlined focus-on-dark' href={process.env.DEFAULT_URL} onClick={backToDashboard}>Back to my dashboard</a>
    </div>
  </div>)
}

export default ThankYouSlide
