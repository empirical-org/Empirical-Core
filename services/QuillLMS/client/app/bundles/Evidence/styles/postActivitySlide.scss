.post-activity-slide-container {
  background-color: #005341;
  min-height: calc(100% - 60px);
  display: flex;
  flex-direction: column;
  align-items: center;

  .responses-exemplars-container {
    display: flex;
    flex-direction: column;
    align-items: center;
    background-color: #FFFFFF;
    width: 100%;
    max-width: 980px;
    margin-bottom: 40px;
    border-radius: 6px;
    border: solid 1px #e0e0e0;
    .review-response-header-section {
      margin-top: 40px;
      max-width: 728px;
      width: 100%;
      text-align: center;
      border-bottom: 1px solid #e0e0e0;
      .sub-header-subtext {
        margin-bottom: 40px;
      }
    }
    .response-exemplars-section {
      display: flex;
      margin-top: 16px;
      border-bottom: 1px solid #e0e0e0;
      .response-section {
        margin-right: 16px;
        .response-exemplar-header {
          display: flex;
          justify-content: center;
          margin-bottom: 8px;
        }
      }
      .exemplars-section {
        margin-left: 16px;
        .response-exemplar-header {
          text-align: center;
          margin-bottom: 8px;
        }
        .exemplar-box {
          p {
            u {
              margin-right: 4px;
            }
          }
        }
        .exemplar-box:last-of-type {
          margin-bottom: 80px;
        }
      }
    }
    .exemplar-box, .response-box {
      width: 450px;
      margin-bottom: 40px;
      background-color: #f5f5f5;
      padding: 16px;
      border-radius: 6px;
    }
  }

  #information-section {
    display: flex;
    flex-direction: column;
    align-items: center;
    justify-content: center;
    flex: 1;
    .slide-sub-text {
      font-size: 16px;
      font-weight: 600;
      color: #b4e2db;
      max-width: 552px;
      text-align: center;
    }

    #revision-text {
      text-align: center;
      font-size: 40px;
      font-weight: bold;
      color: #ffffff;
      max-width: 616px;
      margin-bottom: 8px;
      line-height: 1.2;
    }

    #celebration-vector {
      margin: 16px 0px 16px 0px;
    }

    #reminder-badge-section {
      display: flex;
      align-items: center;
      width: 440px;
      background-color: #e1f3f1;
      margin: 32px 0px 26px 0px;
      border-radius: 8px;
      box-shadow: 0 5px 5px -3px rgba(0, 0, 0, 0.2), 0 3px 14px 2px rgba(0, 0, 0, 0.12), 0 8px 10px 1px rgba(0, 0, 0, 0.14);
      transform: rotate(-2deg);
      #reminder-text-section {
        padding: 16px 0px;
        .sub-header-subtext {
          max-width: 352px;
        }
      }
      #grade-badge {
        margin: 0px 16px 0px 16px;
        width: 48px;
      }
    }
  }

  .sub-header-text {
    color: #000000;
    font-weight: bold;
  }
  .sub-header-subtext {
    color: #262626;
  }

  #button-container {
    display: flex;
    justify-content: center;
    align-items: center;
    border-top: 1px solid white;
    width: 100%;
    flex: 0 0 auto;
    height: 80px;
  }
}


@media (max-width: 992px) {
  .post-activity-slide-container {
    height: calc(100% - 48px);
    overflow-x: hidden;
    width: 100%;

    #information-section {

      .slide-sub-text {
        text-align: center;
        padding: 0px 16px;
        font-size: 20px;
      }

      #revision-text {
        text-align: center;
        font-size: 24px;
        margin: 0px 16px 0px 16px;
      }

      #celebration-vector {
        max-width: 320px;
      }

      #reminder-badge-section {
        width: 280px;
        padding: 0px 16px;
        margin: 0px 38px 24px 38px;

        #grade-badge {
          display: none;
        }
      }
      #second-sub-text {
        margin-top: 22px;
        padding-bottom: 16px;
      }
      #first-sub-text {
        margin-top: 32px;
      }
    }

    .responses-exemplars-container {
      width: 90%;
      margin-bottom: 96px;
      .review-response-header-section {
        .sub-header-subtext {
          margin: 0px 16px 16px 16px;
        }
        .sub-header-text {
          text-align: center;
        }
      }
      .response-exemplars-section {
        flex-direction: column;
        align-items: center;
        .response-section, .exemplars-section {
          margin: 0;
          padding: 0px 16px 0px 16px;
          .response-box, .exemplar-box {
            width: 100%;
          }
          .exemplar-box:last-of-type {
            margin-bottom: 40px
          }
        }
      }
    }

    #button-container {
      justify-content: flex-start;
      position: fixed;
      bottom: 0;
      background-color: #025341;
      button {
        margin-left: 16px;
      }
    }
  }
}

@media (max-height: 670px) {
  .post-activity-slide-container {
    height: min-content;
    padding-top: 16px;
  }
}
