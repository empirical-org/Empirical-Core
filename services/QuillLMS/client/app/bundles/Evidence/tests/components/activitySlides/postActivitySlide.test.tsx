import { shallow } from 'enzyme';
import * as React from 'react';

import { BECAUSE, BUT, SO } from '../../../../Shared/index';
import { PostActivitySlide } from '../../../components/activitySlides/postActivitySlide';

describe('PostActivitySlide Component', () => {
  const mockProps = {
    responses: {
      1: [{ entry: 'test1' }],
      2: [{ entry: 'test2' }],
      3: [{ entry: 'test3' }]
    },
    prompts: [{conjunction: BECAUSE, text: 'test'}, {conjunction: BUT, text: 'test'}, {conjunction: SO, text: 'test'}],
    handleClick: jest.fn()
  }
  let component = shallow(<PostActivitySlide {...mockProps} />);

  it('should match snapshot', () => {
    expect(component).toMatchSnapshot();
  });
});
