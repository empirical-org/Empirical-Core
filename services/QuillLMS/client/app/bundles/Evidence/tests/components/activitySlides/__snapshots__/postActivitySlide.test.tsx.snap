// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PostActivitySlide Component should match snapshot 1`] = `
<div
  className="post-activity-slide-container no-focus-outline"
  tabIndex={-1}
>
  <section
    id="information-section"
  >
    <img
      alt="An illustration of a party popper"
      id="celebration-vector"
      src="undefined/images/evidence/party-celebration.svg"
    />
    <p
      id="revision-text"
    >
      You have completed the activity!
    </p>
    <p
      className="slide-sub-text"
      id="second-sub-text"
    >
      Be proud of the work you did today, and celebrate your success! This practice will help you grow as a reader and a writer. The more you practice, the stronger your critical thinking, reading, and writing skills will be.
    </p>
    <section
      id="reminder-badge-section"
    >
      <img
        alt="An illustration of an A+ that is crossed out"
        id="grade-badge"
        src="undefined/images/evidence/paper-check.svg"
      />
      <section
        id="reminder-text-section"
      >
        <p
          className="sub-header-text"
        >
          Reminder about grades
        </p>
        <p
          className="sub-header-subtext"
        >
          This is practice, so your teacher will see your revisions, but Quill won't assign you a grade.
        </p>
      </section>
    </section>
  </section>
  <section
    className="responses-exemplars-container"
  >
    <section
      className="review-response-header-section"
    >
      <p
        className="sub-header-text"
      >
        Reflect on your work
      </p>
      <p
        className="sub-header-subtext"
      >
        There are many different ways to use evidence in your writing. Consider how your responses compare to the strong examples. In what ways are they similar or different? Notice the ideas, the phrasing, and the tone of voice, and think about how you could use because, but, and so in your future writing.
      </p>
    </section>
    <section
      className="response-exemplars-section"
    >
      <section
        className="response-section"
      >
        <div
          className="response-exemplar-header"
        >
          <p
            className="sub-header-text"
          >
            Your response
          </p>
        </div>
        <section
          className="response-box"
        >
          <p>
            test1
            <strong>
               because 
            </strong>
            <br />
            <u />
          </p>
        </section>
      </section>
      <section
        className="exemplars-section"
      >
        <div
          className="response-exemplar-header"
        >
          <p
            className="sub-header-text"
          >
            Example strong responses
          </p>
        </div>
        <section
          className="exemplar-box"
        >
          <p>
            test
            <strong>
              because 
            </strong>
            <br />
            <u />
          </p>
        </section>
        <section
          className="exemplar-box"
        >
          <p>
            test
            <strong>
              because 
            </strong>
            <br />
            <u />
          </p>
        </section>
      </section>
    </section>
    <section
      className="response-exemplars-section"
    >
      <section
        className="response-section"
      >
        <div
          className="response-exemplar-header"
        >
          <p
            className="sub-header-text"
          >
            Your response
          </p>
        </div>
        <section
          className="response-box"
        >
          <p>
            test2
            <strong>
               but 
            </strong>
            <br />
            <u />
          </p>
        </section>
      </section>
      <section
        className="exemplars-section"
      >
        <div
          className="response-exemplar-header"
        >
          <p
            className="sub-header-text"
          >
            Example strong responses
          </p>
        </div>
        <section
          className="exemplar-box"
        >
          <p>
            test
            <strong>
              but 
            </strong>
            <br />
            <u />
          </p>
        </section>
        <section
          className="exemplar-box"
        >
          <p>
            test
            <strong>
              but 
            </strong>
            <br />
            <u />
          </p>
        </section>
      </section>
    </section>
    <section
      className="response-exemplars-section"
    >
      <section
        className="response-section"
      >
        <div
          className="response-exemplar-header"
        >
          <p
            className="sub-header-text"
          >
            Your response
          </p>
        </div>
        <section
          className="response-box"
        >
          <p>
            test3
            <strong>
               so 
            </strong>
            <br />
            <u />
          </p>
        </section>
      </section>
      <section
        className="exemplars-section"
      >
        <div
          className="response-exemplar-header"
        >
          <p
            className="sub-header-text"
          >
            Example strong responses
          </p>
        </div>
        <section
          className="exemplar-box"
        >
          <p>
            test
            <strong>
              so 
            </strong>
            <br />
            <u />
          </p>
        </section>
        <section
          className="exemplar-box"
        >
          <p>
            test
            <strong>
              so 
            </strong>
            <br />
            <u />
          </p>
        </section>
      </section>
    </section>
  </section>
  <section
    id="button-container"
  >
    <button
      className="quill-button large secondary outlined focus-on-dark"
      onClick={[MockFunction]}
      type="button"
    >
      Next
    </button>
  </section>
</div>
`;
